provider-service/vai/internal/provider/label_sanitizer.go

```diff
@@ -0,0 +1,52 @@
+package provider
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/sky-uk/kfp-operator/provider-service/vai/internal/label"
+)
+
+type LabelSanitizer interface {
+	Sanitize(labels map[string]string) map[string]string
+}
+
+type DefaultLabelSanitizer struct{}
+
+// Vertex AI PipelineJob label keys and values can be no longer than 64 chars,
+// can only contain lowercase letters, numeric chars, underscore and hyphens.
+var vaiCompliant = regexp.MustCompilePOSIX(`[^a-z0-9_-]+`)
+
+// Sanitize returns a processed copy of labels which are sanitized to be compliant
+// to Vertex AI PipelineJob's labels restrictions.
+func (dls DefaultLabelSanitizer) Sanitize(labels map[string]string) map[string]string {
+	const maxLength = 63
+	sanitized := make(map[string]string, len(labels))
+
+	for kSan, vSan := range labels {
```

Comment:
kSan/vSan doesn't make much sense to me, how about:
```suggestion
	for labelKey, labelValue := range labels {
```